src/lib.rs

```diff
@@ -381,25 +382,27 @@ where
     }
 }
 
-impl<T, P> AtomicPtr<T, domain::Global, P> {
+impl<T, F, P> AtomicPtr<T, F, P>
+where
+    F: Singleton,
+{
     /// Loads the value from the stored pointer and guards it using the given hazard pointer.
     ///
     /// The guard ensures that the loaded `T` will remain valid for as long as you hold a reference
     /// to it.
     ///
-    /// Note that this method is _only_ available when using [`Domain<Global>`](domain::Global),
-    /// since it is a singleton, and thus is guaranteed to fulfill the safety requirement of
-    /// [`AtomicPtr::load`]. For all other domains, use [`AtomicPtr::load`].
-    pub fn safe_load<'hp>(
-        &'_ self,
-        hp: &'hp mut HazardPointer<'static, domain::Global>,
-    ) -> Option<&'hp T>
+    /// It's safe for domains with __singleton families__, because the
+    /// [`Domain::Singleton`](unsafety of the trait) guarantees that there is only ever one
+    /// instance of a Domain with a singleton family, and therefore loads and stores using such a
+    /// family must be using the same (single) instance of that domain.
```

Comment:
```suggestion
    /// This method is only available for domains with _singleton families_, because
    /// implementations of the unsafe [`Domain::Singleton`] trait guarantee that there
    /// is only ever one instance of a Domain with the family in question, which in turn
    /// implies that loads and stores using such a family **must** be using the same
    /// (single) instance of that domain.
```